### PR TITLE
ref(useOverlay): Add `maxSize` modifier

### PR DESCRIPTION
**Pull request by vuluongj20**
Does the same thing as [Floating UI's `size` middleware](https://floating-ui.com/docs/size):
https://user-images.githubusercontent.com/44172267/202562953-ac74d307-3944-42d0-839b-661c1d1da3b7.mov

### CODE REVIEW

File: static/app/utils/useOverlay.tsx [hunk 1]
```diff
@@ -11,6 +12,63 @@ import {mergeProps} from '@react-aria/utils';
...
 import {OverlayTriggerProps as OverlayTriggerStateProps} from '@react-types/overlays';
 
+type PreventOverflowOptions = NonNullable<typeof preventOverflow['options']>;
+
+/**
+ * PopperJS modifier to change the popper element's width/height to prevent
+ * overflowing. Based on
+ * https://github.com/atomiks/popper.js/blob/master/src/modifiers/maxSize.js
+ */
+const maxSize: Modifier<'maxSize', PreventOverflowOptions> = {
```
Comment (vuluongj20) [inline static/app/utils/useOverlay.tsx:22]: For now while we're still on `popperjs` we'll have to implement this manually. Once we switch over to Floating UI, we can use it's native `size` middleware.

File: static/app/utils/useOverlay.tsx [hunk 2]
```diff
@@ -109,11 +169,19 @@ function useOverlay({
         enabled: true,
         options: {
           padding: 16,
-          altAxis: true,
```
Comment (vuluongj20) [inline static/app/utils/useOverlay.tsx:112]: With `maxSize` we no longer need to prevent overflow on the alt axis.

**Review by malwilley — APPROVED** [review]
This is pretty cool

**Review by eliashussary — APPROVED** [review]
Looks great!

**Pull request merged**